### ISSUE TITLE
A NAN bug when using SumSquaredDiffs  cost function

### ISSUE THREAD

**Issue by xiuxiazhang** [NervanaSystems/neon#71]
Hello, 
          A NAN bug  occurs when using SumSquaredDiffs as loss function when I am  training own my
          model.   Also, I found this bug  not only occurs to my own kernel. When I  test alex-net example,   and change the cost function to SumSquaredDiffs. It also has   the nan bug in the first epoch.

           In the yaml file, the only different I made is change CrossEntropy to SumSquaredDiffs.

        In my model,  NAN error occurs after about 55 fprop  and bprop  in fit funciton.
        It seems some value becomes INF after several iterations.
             
        Are there  functions that may generate INF value when using sumsquareddiffs cost function?

Thanks,
Xiuxia

**Comment by scttl** [comment]
Hi Xiuxia,

This isn't a bug -- in general you can't expect to just swap one cost function for another without also adjusting or rescaling the other network parameters.

There are any number of ways you can get into the numerical issues you're seeing, so I can't really provide specifics for your situation.  Your best bet for getting to the bottom of them is to use a debugger and monitor some of the tensors in each layer to see if their values are blowing up, or getting arbitrarily close to zero (which could yield INF values when acting as the denominator in a division operation like what happens during backprop).
